src/execution_tree/primitives/slice.cpp

```diff
@@ -133,12 +138,18 @@ namespace phylanx { namespace execution_tree
                         std::move(value), name, codename),
                     name, codename)};
             }
+        } 
+        if (is_dictionary_operand(data))
+        {
+            auto f = phylanx::execution_tree::extract_dictionary_value(data);
+            f[indices] = value;
+            return primitive_argument_type{f};
```

Comment:
I would have expected for similar code to be necessary in the variable primitive. How is
```
@Phylanx
def dc2():
    a = {1:2, 3:4}
    a[1] = 5
    return a
```
translated to PhySL? For best possible performance I'd like for `a[1] = 5` to become `store(slice(a, 1, nil), 5)`. The PhySL compiler will transform this into a (extended) call to `variable::store` (see: https://github.com/STEllAR-GROUP/phylanx/blob/master/src/execution_tree/primitives/store_operation.cpp#L176).

Reply:
It gets translated as "store(slice(a,1), 5)". There really isn't anything that can be done about that, since we can't have the translation depend on the variable type.

Reply:
That's fine.